src/utils/Helper.ts

```diff
@@ -21,36 +23,108 @@ export function addModelToFormData(model: any, form: FormData, namespace = ''):
   return form
 }
 
-export function removeDuplicates<T>(array: T[]): T[] {
+function removeDuplicates<T>(array: T[]): T[] {
   return Array.from(new Set(array))
 }
 
 /**
- * This method performs splits what would be one query into many, to avoid exceeding the max length of urls
+ * This method performs one or more fetches to the given server, splitting query params into different queries to avoid exceeding the max length of urls
  */
 export const MAX_URL_LENGTH: number = 2048
-export function splitValuesIntoManyQueries(
-  baseUrl: string,
-  basePath: string,
-  queryParamName: string,
-  values: string[]
-): string[] {
-  return splitManyValuesIntoManyQueries(baseUrl, basePath, new Map([[queryParamName, values]]))
+export async function splitAndFetch<E>({
+  baseUrl,
+  path,
+  queryParams,
+  fetcher,
+  uniqueBy,
+  options
+}: Omit<SplitAndFetchParams<E>, 'elementsProperty'>): Promise<E[]> {
+  // Adding default
+  fetcher = fetcher ?? new Fetcher()
+
+  // Split values into different queries
+  const queries = splitValuesIntoManyQueries({ baseUrl, path, queryParams })
+
+  const results: Map<any, E> = new Map()
+  for (const query of queries) {
+    // Perform the different queries
+    const elements: E[] = await fetcher.fetchJson(query, options)
+
+    // Group by unique property (if set), or add all of them to the map
+    elements.forEach((element) => results.set(uniqueBy ? element[uniqueBy] : results.size, element))
```

Comment:
As long as I understood each query represents a different query param, this means that the logic of using many query parameters is equivalent to the `OR` logic operator?

Reply:
It depends. If you for example have:
```
filter = {
    wearableIds: ['id1', 'id2' ]
    collectionIds: ['cid1', 'cid2']
}
```
it translates to:
```
(wearableId == 'id1' OR wearableId == 'id2') AND (collectionId == 'cid1' OR collectionId == 'cid2')
```
The logic is:
* AND between different params
* OR between values of the same param